pkg/cluster/k8sres.go

```diff
@@ -431,6 +435,33 @@ func (c *Cluster) generatePodTemplate(resourceRequirements *v1.ResourceRequireme
 		Tolerations:                   c.tolerations(tolerationsSpec),
 	}
 
+	if c.OpConfig.ScalyrAPIKey != "" && c.OpConfig.ScalyrImage != "" {
+		podSpec.Containers = append(
+			podSpec.Containers,
+			v1.Container{
+				Name:            "scalyr-sidecar",
+				Image:           c.OpConfig.ScalyrImage,
+				ImagePullPolicy: v1.PullIfNotPresent,
+				Resources:       *resourceRequirementsScalyrSidecar,
+				VolumeMounts:    volumeMounts,
```

Comment:
Why is there a need to mount the postgres data volume to the scalyr agent container?

Reply:
Because it needs to watch the log files for changes, and the log files are in the Postgres data volume.  We could also put the log files in a separate volume, but I don't know if we want to do this.